src/sentry/api/endpoints/project_performance_issue_settings.py

```diff
@@ -63,14 +63,26 @@ def get(self, request: Request, project) -> Response:
         if not self.has_feature(project, request):
             return self.respond(status=status.HTTP_404_NOT_FOUND)
 
-        performance_issue_settings_default = projectoptions.get_well_known_default(
+        system_defaults = {
+            "n_plus_one_db_duration_threshold": options.get(
+                "performance.issues.n_plus_one_db.duration_threshold"
+            ),
+            "slow_db_query_duration_threshold": options.get(
+                "performance.issues.slow_db_query.duration_threshold"
+            ),
+        }
+
+        project_performance_settings_defaults = projectoptions.get_well_known_default(
             SETTINGS_PROJECT_OPTION_KEY,
             project=project,
         )
-        performance_issue_settings = project.get_option(
-            SETTINGS_PROJECT_OPTION_KEY, default=performance_issue_settings_default
+        project_performance_settings = project.get_option(
+            SETTINGS_PROJECT_OPTION_KEY, default=project_performance_settings_defaults
         )
-        return Response({**performance_issue_settings_default, **performance_issue_settings})
+
+        project_settings = {**project_performance_settings_defaults, **project_performance_settings}
+
+        return Response({**system_defaults, **project_settings})
```

Comment:
This is basically mimicking the option merging in the ingest detection code, right?

https://github.com/getsentry/sentry/blob/0b2dd74a670392e8c612289417a4a4a741873921/src/sentry/utils/performance_issues/performance_detection.py#L182-L187

I'm wondering if we maybe can pull that out and share the final option bundle after merging defaults + system + project.

Reply:
Yeah, correct they do the same thing. I'll pull the merged settings out and share it between them in the next PRs.